refactor: simplify Dockerfile.vllm, enable local-dev for all frameworks

keivenchang opened the pull request:
#### Overview:

Refactor container build system to enable local-dev images for all frameworks while simplifying Dockerfile.vllm.

#### Details:

- Extract local-dev stage from Dockerfile.vllm to new Dockerfile.local_dev
- Add build_local_dev.sh script for converting dev images to local-dev images
- Enable local-dev builds for all frameworks via unified approach
- Update documentation with two-step build process

#### Where should the reviewer start?

container/Dockerfile.local_dev and container/build_local_dev.sh for new infrastructure, then container/Dockerfile.vllm for removed complexity.

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

/coderabbit disable

keivenchang committed:
refactor: simplify Dockerfile.vllm, enable local-dev for all frameworks

- Move local-dev target to new Dockerfile.local_dev
- Add build_local_dev.sh script for converting dev images to local-dev images
- Redirect local-dev builds to new dedicated script
- Enable local-dev builds for all frameworks via unified approach
- Remove complexity by extracting local-dev stage
- Update documentation with new two-step build process

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
docs: update troubleshooting section for new local-dev build process

- Update missing image section to reflect local-dev image requirement
- Add two-step build process with framework example (vllm)
- Show expected output image name (dynamo:latest-vllm-local-dev)
- Clarify troubleshooting steps for new build_local_dev.sh workflow

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

nv-tusharma commented on container/build_local_dev.sh:
```diff
@@ -0,0 +1,278 @@
+
+
+# Default values
+ARCH="amd64"
```
Should be defined by the architecture of the base image

nv-tusharma commented on container/build_local_dev.sh:
```diff
@@ -0,0 +1,278 @@
+    echo "[ERROR] $1"
+}
+
+list_dev_images() {
```
What is the use case for this function?

keivenchang committed:
feat: integrate local-dev build into build.sh and improve documentation

- Add --target local-dev support to build.sh (builds both dev and local-dev images)
- Refactor build_local_dev.sh to use RUN_PREFIX and consistent tag handling
- Add --squash option to build_local_dev.sh for smaller images
- Update documentation to clarify dev vs local-dev image usage
- Improve terminology: dev images run as root, local-dev run as local user
- Add guidance on when to use each image type (mounting vs inference)
- Streamline examples to show single-command approach with build.sh

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
refactor: consolidate local-dev build logic into build.sh

- Add --dev-image option to build from existing dev images
- Add --uid and --gid options for custom user permissions
- Consolidate duplicate docker build logic with shared function
- Add validation to prevent conflicting --target and --dev-image options
- Fix tag generation to always add -local-dev suffix
- Add dynamic path calculation for usage instructions
- Rename LOCAL_DEV_BASE to DEV_BASE for clarity
- Remove unused options from build_local_dev.sh
- Update terminology from 'converting' to 'building'

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
remove: delete build_local_dev.sh after consolidation

All functionality has been moved to build.sh with improved integration

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
docs: replace build_local_dev.sh references with build.sh commands

- Update all documentation to use build.sh --dev-image instead of build_local_dev.sh
- Simplify container README section for local development image builder
- Remove references to deleted build_local_dev.sh script
- Update example commands to use consolidated build.sh interface

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>

keivenchang committed:
docs: clarify dev tools installation comment in local_dev Dockerfile

Signed-off-by: Keiven Chang <keivenchang@users.noreply.github.com>
